File: src/main/scala/test/TestScala.java
package test;

//object TestScala运行的一个具体流程（反编译过后）
public class TestScala {
    public static void main(String[] args) {
        TestScala$.Module$.main(args);
    }
}

final class TestScala${
    public static final TestScala$ Module$;

    static {
        Module$ = new TestScala$();
    }

    public static void main(String[] args) {
        System.out.println("hello scala~~");
    }
}

File: src/main/scala/TestScala.scala

//1.我们可以理解scala再运行时进行了一个包装
//2.object TestScala 对应的是一个TestScala$的静态单例对象Module$
object TestScala {

  def main(args: Array[String]): Unit = {
    println("hello,world~~")
  }

}
